DR_0011_React-for-paginated-views.md

```diff
@@ -0,0 +1,194 @@
+
+We also support basic search and sorting sorting, which is sufficient to implement client-side searchable and sortable lists. 
+
+Aside from that, we might have to add smaller changes here and there to fully support the requirements for each of the paginated views.
+
+#### Client-side extensions
+
+We will implement a set of React components that will allow us to compose paginated tables that mirror the functionality currently achieved using DataTables.
+
+The React component `PaginatedTable` will be use across all paginated views and will provide the core functionality including search, pagination navigation, setting the page size and column sorting. `PaginatedTable` will be the root component and will host the store and all event handlers. Functionality listed above will be implemented as separate React components. 
+
+##### Configuring `PaginatedTable`
+
+`PaginatedTable` is configurable. Each paginated view has a different API source (where we load the data from), number of columns and template for how each row in the table should be rendered. 
+
+The following properties must be set when initialising `PaginatedTable`:
+
+| Property         | Type              | Description   | 
+| ---------------- | ----------------- | ------------- | 
+| `url`            | `String`          | The API URL where data is loaded from. | 
+| `columns `       | `Array`           | A list of column definitions. The number of elements defines the number of columns in the table. Each element must define the following keys: <br>`label` (the label used in the table head for the column)<br>`field` (the model field, used to apply sorting)<br>`sortable` (identifies if the column is sortable) keys. | 
+| `row_component ` | `React.Component` | A React component that is used to render individual rows into the table. This is necessary because so tables (such as resources) require more complex rendering of individual cells. | 
+
+##### Example use
+
+```html
+<!-- Load React libs -->
+<script crossorigin src="https://unpkg.com/react@15/dist/react.min.js"></script>
+<script crossorigin src="https://unpkg.com/react-dom@15/dist/react-dom.min.js"></script>
+
+<!-- Load the PaginateTable core component -->
+<script src="/static/dist/PaginatedTable.min.js"></script>
+
+<!-- Load the row component used in this view -->
+<script src="/static/dist/PartyRow.min.js"></script>
+<script type="text/javascript">
+  var props = {
+    "url": "/api/v1/resources",
+    "columns": [
+      {"label": "Party", "field": "name", "sortable": true},
+      {"label": "Type", "field": "type", "sortable": true}
+    ],
+    "row_component": Row
+  }
+  ReactDOM.render(
+    React.createElement(PaginatedTable, props),
+    document.getElementById('paginated-table')
+  );
+</script>
+```
+
+A very basic implementation can be found in [branch `react-party`](https://github.com/Cadasta/cadasta-platform/compare/react-party).
```

Comment:
I think it's best if we avoid linking to branches in DRs, as I don't have confidence that they'll persist for as long as the DR will.

Looking over the example, it doesn't actually seem to address the feature of pagination. Is there any reason we're not starting off of [the prototype](https://github.com/Cadasta/cadasta-platform/commit/2ad8f9288625325c67b7c67fc946ad11fb7e132b?diff=unified) that I put together?  It seems like the example contains a subset of features.